hw/ip/csrng/data/csrng.hjson

```diff
@@ -95,18 +95,46 @@
     { name: "INTERSIG.MUBI"
       desc: "OTP signal used to enable software access to registers."
     }
-    { name: "FSM.SPARSE"
-      desc: "Sparse state machine implementation."
+    { name: "MAIN_SM.FSM.SPARSE"
+      desc: "The CSRNG main state machine uses a sparse state encoding."
     }
-    { name: "CTR.REDUN"
-      desc: "Counter hardening for generate command counter."
+    { name: "UPDATE.FSM.SPARSE"
+      desc: "The CSRNG update state machine uses a sparse state encoding."
+    }
+    { name: "BLK_ENC.FSM.SPARSE"
+      desc: "The CSRNG block encrypt state machine uses a sparse state encoding."
+    }
+    { name: "OUTBLK.FSM.SPARSE"
+      desc: "The CSRNG block output state machine uses a sparse state encoding."
+    }
+    { name: "GEN_CMD.CTR.REDUN"
+     desc: "The generate command uses a counter that is protected by a second counter that counts in the opposite direction."
+    }
+    { name: "DRBG_UPD.CTR.REDUN"
+     desc: "The ctr_drbg update algorthm uses a counter that is protected by a second counter that counts in the opposite direction."
+    }
+    { name: "DRBG_GEN.CTR.REDUN"
+     desc: "The ctr_drbg generate algorthm uses a counter that is protected by a second counter that counts in the opposite direction."
+    }
+    { name: "CTRL.MUBI"
+      desc: "Multi-bit field used for selection control."
+    }
+    { name: "MAIN_SM.CTR.LOCAL_ESC"
+      desc: "A mismatch detected inside any CSRNG counter moves the  main state machine into a terminal error state."
+    }
+    { name: "CONSTANTS.LC_GATED"
+      desc: "Seed diversification based on the lifecycle state."
+    }
+    { name: "FSM.LC_GATED"
+      desc: "Debug state machines are gated with lifecycle state."
     }
     { name: "SW_GENBITS.BUS.CONSISTENCY"
       desc: "Comparison on successive bus values for genbits returned on the software channel."
     }
     { name: "TILE_LINK.BUS.INTEGRITY"
       desc: "Tilelink end-to-end bus integrity scheme."
     }
+    // See the AES module documentation for additional countermeasures provided by that design"
```

Comment:
Sorry, I probably didn't explain clearly. What I would do is something like:
```
    { name: "AES_CIPHER.FSM.SPARSE"
      desc: "See the AES module documentation for details."
    }
    { name: "AES_CIPHER.CTRL.REDUN"
      desc: "See the AES module documentation for details."
    }
    ...
    { name: "AES_CIPHER.DATA_REG.LOCAL_ESC"
      desc: "See the AES module documentation for details."
    }
```
The reason for explicitly putting the labels is that some countermeasures inside AES are parametrized and not enabled for the AES instantiation inside CSRNG. And I think we should document that here.

But I wouldn't repeat the `desc` field here. For a detailed description people should look up the AES documentation. Otherwise we have to make sure to keep the desc fields aligned, too.

Reply:
@msfschaffner @vogelpi hmmm.... this brings to mind.
I wonder if we need a reference category handle on the names here. 

Because let's say AES had 100 countermeasures (it's not that far off in reality), we wouldn't really want `csrng` to repeat them everywhere all the time...and any rtl update to change two places right? 

It would be good if in csrng we could instead say "we have these categories of countermeasures from aes" based on however the parameter is set. 

As an example right now, when we check countermeasure existence for csrng, we probably don't check the aes files do we? (since it's not really based on fusesoc I thought).